app/experimenter/visualization/api/v3/views.py

```diff
@@ -68,7 +68,45 @@ def get_results_metrics_map(primary_outcomes, secondary_outcomes):
     for outcome_slug in secondary_outcomes:
         RESULTS_METRICS_MAP[outcome_slug] = set([Statistic.MEAN])
 
-    return RESULTS_METRICS_MAP, primary_metrics_set
+    other_metrics_map, other_metrics = get_other_metrics_names_and_map(
+        data, RESULTS_METRICS_MAP
+    )
+    RESULTS_METRICS_MAP.update(other_metrics_map)
+
+    return RESULTS_METRICS_MAP, primary_metrics_set, other_metrics
+
+
+def get_other_metrics_names_and_map(data, RESULTS_METRICS_MAP):
+    # These are metrics sent from Jetstream that are not explicitly chosen
+    # by users to be either primary or secondary
+    other_metrics_names = {}
+    other_metrics_map = {}
+
+    # This is an ordered list of priorities of stats to graph
+    priority_stats = [Statistic.MEAN, Statistic.BINOMIAL]
+    other_data = [
+        data_point
+        for data_point in data
+        if data_point["metric"] not in RESULTS_METRICS_MAP
+    ]
+    for row in other_data:
+        metric = row.get("metric")
+        statistic = row.get("statistic")
```

Comment:
I guess the only thing to think about is the case where those keys aren't present and return `None`, not sure if that'll break things?